design/design.md

```diff
@@ -0,0 +1,380 @@
+Once the second flow is created, it does not wait for the first flow to finish,
+or synchronize with the first flow in any way,
+and could potentially finish executing before the first flow finishes.
+
+There will be a handler for each gRPC method, separate from the existing json
+handlers. The handler will take in the protobuf object that represents the
+request, and return a protobuf object that represents the response. The protobuf
+object will be a member of a templated class `RPC::ContextGeneric<T>`, where T
+is the type of the protobuf object. `RPC::ContextGeneric<T>` is very similar to
+`RPC::Context`, except the `params` member is of type `T` and there are no
+`Headers`.
+
+Below is an example handler signature; `xrp::v1` is the namespace of the
+protobuf objects, whereas `GetFeeRequest` is the request type and `Fee` is the
+response type.
+```
+xrp::v1::Fee doFee(RPC::ContextGeneric<xrp::v1::GetFeeRequest>& context);
+```
+Versioning for gRPC is usually done by changing the package name, which creates
+protobuf objects in a new c++ namespace. For example, changing the package name
+to `xrp.v2` results in objects with namespace `xrp::v2`. In the event
+that a handler does not change between versions (which implies the objects are
+the same in all but namespace), the old handler can be reused by templating the
+handler signature like so:
+
+```
+template<class T, class R>
+R doFee(RPC::ContextGeneric<T>& context);
+```
+
+## Interface
+The interface to the GRPC service is defined by the .proto files. The `.proto`
+files define the structure of the messages, the methods available, their inputs
+and outputs, and the service itself.  GRPC, with the help of protobuf,
+autogenerates code based on these `.proto` files, for both client and server.
+GRPC has bindings in many common programming languages
+([link](https://grpc.io/docs/reference/)).
+A client of the GRPC service we are building would use the exact same `.proto`
+files as the server, to generate the client code. The generated code (both
+client and server), handles serialization, deserialization and I/O. Protobuf
+allows one to define messages with named and typed fields, and allows for user
+defined types. Client and server code can then manipulate the generated objects using the
+message names, field names and types defined in the `.proto` file. For example,
+using the below example `.proto` file:
+```
+service XRPLedgerAPI {
+  // Get account info for an account on the XRP Ledger.
+  rpc GetAccountInfo (GetAccountInfoRequest) returns (AccountInfo);
+
+  // Get the fee for a transaction on the XRP Ledger.
+  rpc GetFee (GetFeeRequest) returns (Fee);
+  message GetAccountInfoRequest {
+    // The address to get info about.
+    bytes address = 1;
+  }
+
+  message AccountInfo {
+    XRPAmount balance = 1;
+  
+    uint32 sequence = 2;
+  
+  }
+  message XRPAmount {
+    // A numeric string representing the number of drops of XRP.
+    bytes drops = 1;
```

Comment:
I see we're using a string interface for this. Why not a binary interface (i.e. `uint64`)?

Reply:
changed to uint64